src/web/extractors.rs

```diff
@@ -530,14 +539,22 @@ impl CollectionParam {
             return Ok(None);
         }
         if let Some(v) = elements.get(4) {
-            let sv = String::from_str(v).map_err(|_e| {
+            let mut sv = String::from_str(v).map_err(|_e| {
                 ValidationErrorKind::FromDetails(
                     "Missing Collection".to_owned(),
                     RequestErrorLocation::Path,
                     Some("collection".to_owned()),
                     Some(tags.clone()),
                 )
             })?;
+            sv = clean_element(&sv).map_err(|_e| {
```

Comment:
It occurs to me now that we can probably now kill significant `extrude` paths that we made to work against `ServiceRequest` now that we've removed the 2 larger middlewares. E.g. all the `x_from_path` methods could work on the regular actix-web `HttpRequest` again, referencing its `match_info()` call to grab these values from the path (and actix would urldecode them for us).

This is a pretty quick fix so that's probably a separate issue 😄

Reply:
Huh, yeah, you're right. 

I think it should be a different issue as well.